test/models/user_test.rb

```diff
@@ -3,7 +3,54 @@
 require 'test_helper'
 
 class UserTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  setup do
+    @user_one = users(:one)
+    @user_two = users(:two)
+    @user_three = users(:three)
+  end
+
+  test 'valid following user' do
+    assert_not @user_one.following?(@user_two)
+
+    @user_one.follow(@user_two)
+
+    assert @user_one.following?(@user_two)
+  end
+
+  test 'invalid following user' do
+    assert_not @user_one.following?(@user_two)
+  end
+
+  test 'valid followed by user' do
+    assert_not @user_two.followed_by?(@user_one)
+
+    @user_one.follow(@user_two)
+
+    assert @user_two.followed_by?(@user_one)
+  end
+
+  test 'invalid followed by user' do
+    assert_not @user_two.followed_by?(@user_one)
+  end
+
+  test 'follow a user' do
+    @user_one.follow(@user_two)
+    assert @user_one.following?(@user_two)
+  end
+
+  test 'unfollow a user' do
+    @user_one.follow(@user_two)
+    assert @user_one.following?(@user_two)
+
+    @user_one.unfollow(@user_two)
+    assert_not @user_one.following?(@user_two)
+  end
+
+  test 'should return name if name exist' do
+    assert_equal @user_three.name_or_email, @user_three.name
```

Comment:
`name_or_email`をテストするのであれば、

```suggestion
    assert_equal @user_three.name, @user_three.name_or_email
```

という順番（期待値, 実際の値）で書くのが正しいです。
さらに、期待値は変数ではなく文字列をそのまま使う方が良いです。

```suggestion
    assert_equal 'Alice', @user_three.name_or_email
```